compiler/src/Language/Bond/Parser.hs

```diff
@@ -240,8 +240,9 @@ struct = do
             [] -> return fields'
             Field {..}:_ -> fail $ "Duplicate definition of the field with ordinal " ++ show fieldOrdinal ++
                 " and name " ++ show fieldName
-      where
-        findDuplicatesBy accessor xs = deleteFirstsBy ((==) `on` accessor) xs (nubBy ((==) `on` accessor) xs)
+
+findDuplicatesBy :: (Eq b) => (a -> b) -> [a] -> [a]
```

Comment:
Minor: this feels like an odd place in the file for such a helper method. Move to near beginning/end?